cmd/geth/chaincmd.go

```diff
@@ -675,8 +711,205 @@ func dump(ctx *cli.Context) error {
 	return nil
 }
 
+func exportSegment(ctx *cli.Context) error {
+	stack, _ := makeConfigNode(ctx)
+	defer stack.Close()
+
+	db := utils.MakeChainDatabase(ctx, stack, false, false)
+	defer db.Close()
+
+	genesisHash := rawdb.ReadCanonicalHash(db, 0)
+	td := rawdb.ReadTd(db, genesisHash, 0)
+	chainConfig, engine, headerChain, err := simpleHeaderChain(db, genesisHash)
+	if err != nil {
+		return err
+	}
+	latest := headerChain.CurrentHeader()
+	if _, ok := engine.(consensus.PoSA); !ok {
+		return errors.New("current chain is not POSA consensus, cannot generate history segment")
+	}
+	if !chainConfig.IsLuban(latest.Number) {
+		return errors.New("current chain is not enable Luban hard fork, cannot generate history segment")
+	}
+
+	var (
+		boundStartBlock      = params.BoundStartBlock
+		historySegmentLength = params.HistorySegmentLength
+	)
+	switch genesisHash {
+	case params.BSCGenesisHash, params.ChapelGenesisHash, params.RialtoGenesisHash:
+		boundStartBlock = params.BoundStartBlock
```

Comment:
Not possible to configure BoundStartBlock based on the flag value on mainnet?

Reply:
yes, mainnet or testnet is hard code, every node should follow the same rules.

Reply:
mainnet, chapel and rialto should have different `BoundStartBlock`, right?